server/src/language-service/opctagListProvider.ts

```diff
@@ -0,0 +1,42 @@
+import { SepticReferenceProvider } from "../septic";
+
+export class OpcTagListProvider {
+    constructor() {}
+
+    public generateOpcTagList(refProvider: SepticReferenceProvider): string {
+        return "";
+    }
+}
+
+export function generateOpcReport(
+    refProvider: SepticReferenceProvider
+): string {
+    let header: string = "ObjectId;ObjectType;ObjectAttribute;OPCTag;";
+    let entries: string[] = [];
+    let opcObjects = refProvider.getObjectsByType(
+        "SopcTvr",
+        "SopcEvr",
+        "SopcCvr",
+        "SopcDvr",
+        "SopcMvr",
+        "SopcProc"
+    );
+    for (let obj of opcObjects) {
+        let objectName = obj.identifier?.id ?? "unknown";
+        let tagAttributes = obj.attributes.filter((attr) =>
+            attr.key.endsWith("Tag")
+        );
+        for (let tagAttr of tagAttributes) {
+            let value = tagAttr.getAttrValue()?.getValue() ?? "";
+            if (
+                value.trim() === "" ||
+                value === "DUMMY_TAG" ||
+                value === "NotUsed"
+            ) {
+                continue;
+            }
+            entries.push(`${objectName};${obj.type};${tagAttr.key};${value};`);
```

Comment:
For configs with two or more SopcProc, will it list them in this order:
SopcProc 1 (encountered first)
All Xvrs belonging to SopcProc1
SopcProc 2 (encountered after SopcProc 1)
All Xvrs belonging to SopcProc2
...etc

Looks like this is the case? If so: good.